pump/storage/storage.go

```diff
@@ -847,15 +855,24 @@ func (a *Append) batchRequest(reqs chan *request, maxBatchNum int) chan []*reque
 }
 
 func (a *Append) writeToValueLog(reqs chan *request) chan *request {
+	ctx, cancel := context.WithCancel(context.Background())
 	done := make(chan *request, a.options.KVChanCapacity)
+	slowChaser := newSlowChaser(a.vlog, 3*time.Second, done)
+	go slowChaser.Run(ctx)
 
 	go func() {
-		defer close(done)
+		defer func() {
+			close(done)
+			cancel()
+		}()
 
 		var bufReqs []*request
 		var size int
 
 		write := func(batch []*request) {
+			slowChaser.WriteLock.Lock()
+			defer slowChaser.WriteLock.Unlock()
```

Comment:
I suggest don't use defer in hotpath

Reply:
`defer` is used here because:

1. `write` is a function more than 30 lines long with at least 4 exit paths, it's error-prone to insert `Unblock`s to the correct places;
1. The overhead introduced by using `defer` is trivial compared to the time it takes to save the binlog and then call `fsync`

Reply:
ok, convincing explanation